observatory-dags/observatory/dags/telescopes/oapen_irus_uk.py

```diff
@@ -40,8 +42,9 @@
     create_cloud_storage_bucket, \
     download_blob_from_cloud_storage, \
     upload_file_to_cloud_storage
-from observatory.platform.utils.telescope_utils import make_dag_id, make_org_id
-from observatory.platform.utils.template_utils import SubFolder, blob_name, telescope_path
+from observatory.platform.utils.telescope_utils import make_dag_id, make_org_id, add_partition_date, list_to_jsonl_gz
+from observatory.platform.utils.template_utils import SubFolder, blob_name, telescope_path, bq_load_partition, \
+    table_ids_from_path
 
 
 class OapenIrusUkRelease(SnapshotRelease):
```

Comment:
The cloud function gets the following error ` download_access_stats_new doi = item['Item_ID']['DOI'] KeyError: 'DOI'`:
```
_function_handler.invoke_user_function(flask.request) File "/env/local/lib/python3.7/site-packages/google/cloud/functions/worker_v2.py", line 268, in invoke_user_function return call_user_function(request_or_event) File "/env/local/lib/python3.7/site-packages/google/cloud/functions/worker_v2.py", line 261, in call_user_function return self._user_function(request_or_event) File "/user_code/main.py", line 62, in download download_access_stats_new(file_path, release_date, username, password, publisher_uuid, geoip_client) File "/user_code/main.py", line 219, in download_access_stats_new doi = item['Item_ID']['DOI'] KeyError: 'DOI'
Traceback (most recent call last): File "/env/local/lib/python3.7/site-packages/google/cloud/functions/worker_v2.py", line 402, in run_http_function result = _function_handler.invoke_user_function(flask.request) File "/env/local/lib/python3.7/site-packages/google/cloud/functions/worker_v2.py", line 268, in invoke_user_function return call_user_function(request_or_event) File "/env/local/lib/python3.7/site-packages/google/cloud/functions/worker_v2.py", line 261, in call_user_function return self._user_function(request_or_event) File "/user_code/main.py", line 62, in download download_access_stats_new(file_path, release_date, username, password, publisher_uuid, geoip_client) File "/user_code/main.py", line 219, in download_access_stats_new doi = item['Item_ID']['DOI'] KeyError: 'DOI'
```

Reply:
Mmm, apparently there are some items without a DOI or some of the other identifiers, I was not aware of this.
I've added a try/except block to get these identifiers in the cloud function. Will have to create a new release though and update this in the telescope.